backends/tfhe-cuda-backend/cuda/src/polynomial/polynomial_math.cuh

```diff
@@ -82,4 +83,37 @@ polynomial_product_accumulate_by_monomial(T *result, const T *__restrict__ poly,
   }
 }
 
+// This method expects to work with num_poly * polynomial_size threads in the
+// grid
+template <typename T>
+__device__ void polynomial_accumulate_monic_monomial_mul_batch(
```

Comment:
I don't think we need the _batch suffix, it is implicit all our entry points work on arrays of values.

Reply:
There are two `polynomial_accumulate_monic_monomial_mul` now:

- the original one that takes a single polynomial as input, and expects to work with `polynomial_size / params::opt` threads per block, 
- and a new one that takes an array of polynomials and expects to have `num_poly * polynomial_size` threads in the entire grid.

I believe we need to differentiate these functions since they should be used in different contexts. Any ideas?

Reply:
Code compiles and works if we just overload both. We can keep them with the same name if you prefer.

Reply:
I would prefer to keep only one, maybe it'd be good to add a comment to the function explaining that depending on the number of threads it can handle 1 or multiple polynomials?

Reply:
Hun! Ok.
The comment is already there. I will try to refactor something to keep only one.

Reply:
Done. 

Now we have a single `polynomial_accumulate_monic_monomial_mul`. I had to lose the `params` template argument from the original method and introduce `polynomial_size` and `coeff_per_thread` as function arguments, since `fp_keyswitch` doesn't have the `params` argument. Also, the base position per thread has to passed as function argument, since it is different for the PBS and `fp_keyswitch`.

Take a look and tell me wdyt.

Reply:
Yes looks good thanks!